sqlglot/tokens.py

```diff
@@ -290,6 +290,7 @@ class TokenType(AutoName):
     UPDATE = auto()
     USE = auto()
     USING = auto()
+    USING_SAMPLE = auto()
```

Comment:
i don't think you need this

Reply:
Two reasons for adding this:

- Easily exclude `USING` from the table alias set
- I wanted to attach the using sample expression in the query modifiers list, because it's supposed to be applied to the resulting set instead of a single table. This is why I added the `as_modifier` flag and checked if we can match USING SAMPLE.

Reply:
using is already not allowed in table alias, so you can just exclude that

i don't understand the as_modifier flag

Reply:
> using is already not allowed in table alias, so you can just exclude that

Ah, I missed this because it's not in `ID_VAR_TOKENS` either and so thought we weren't excluding it.

> i don't understand the as_modifier flag

If we just mapped `USING SAMPLE` to `TokenType.TABLE_SAMPLE`, then the former would be attached to the table, instead of being a query modifier. From an AST perspective this is not correct, which is why I introduced the `as_modifier` flag. Does it make sense?

Reply:
well you shouldn't map using sample, but you can just to match_text right?